File: python-chatdownloader/src/leaderboards/overall.py
"""
The overall leaderboard
"""

from typing import Optional

from _types import UserChatPerformance

from .abstractleaderboard import AbstractLeaderboard


class Overall(AbstractLeaderboard):
    """
    Computes the overall leaderboard (Special + Non-Special).
    """

    @classmethod
    def get_name(cls):
        return 'overall'

    def calculate_score(self,
                        performance: UserChatPerformance) -> Optional[float]:
        return sum(performance.metrics.values())

File: rust/elo/src/leaderboards/overall.rs
/*
The overall leaderboard
*/

use crate::_types::clptypes::UserChatPerformance;
use crate::_types::leaderboardtypes::LeaderboardInnerState;
use crate::leaderboards::leaderboardtrait::AbstractLeaderboard;
use std::collections::HashMap;

use super::Leaderboard;

#[derive(Default, Debug)]
pub struct Overall {
    state: HashMap<String, LeaderboardInnerState>,
}

impl AbstractLeaderboard for Overall {
    fn new() -> Self {
        let mut out = Self {
            state: HashMap::new(),
        };
        out.read_initial_state();
        out
    }

    fn get_name(&self) -> String {
        "overall".to_string()
    }

    fn __get_state(&mut self) -> &mut HashMap<String, LeaderboardInnerState> {
        &mut self.state
    }

    fn calculate_score(&self, performance: &UserChatPerformance) -> Option<f32> {
        return Some(performance.metrics.values().sum());
    }
}

impl Into<Leaderboard> for Overall {
    fn into(self) -> Leaderboard {
        Leaderboard::Overall(self)
    }
}
